woocommerce/woocommerce
# Add category field block

### All Submissions:






### Changes proposed in this Pull Request:

This adds a new category block that is currently a wrapper around the `CategoryField` component.
It also adds a new section to the product template.

Initial work on #37263 
**Note:**
There are still some minor styling issues with the Select tree control, which will be done as follow ups:
- Allow the option to list the options as a comma separated list when list is not expanded.
- Fix popover location when the dropdown menu doesn't have enough room at the bottom and renders above the input field.

https://user-images.githubusercontent.com/2240960/225940450-61af8d5c-6e3d-4b22-8f6b-740736eb8abc.mp4

### How to test the changes in this Pull Request:




1. Build this branch and enable the product blocks feature flag: `block-editor-feature-enabled`
2. Go to **WooCommerce > Products** and select a product with categories ( if not create one using the old UI )
3. The product block editor should load correctly and display a categories field with the selected categories.
4. You should be able to search for categories, select new ones, and remove current ones (similar to how the current category dropdown works).



### Other information:




### FOR PR REVIEWER ONLY:
